official/recommendation/data_download.py

```diff
@@ -0,0 +1,296 @@
+  return func(ratings_file, sort=sort)
+
+
+def generate_train_eval_data(df, original_users, original_items):
+  """Generate the dataset for model training and evaluation.
+
+  Given all user and item interaction information, for each user, first to sort
+  the interactions based on timestamp. Then the latest one is taken out as
+  Test ratings (leave-one-out evaluation) and the remaining data for training.
+  The Test negatives are randomly sampled from all non-interacted items, and the
+  number of Test negatives is 100 (defined as NUMBER_NEGATIVES).
+
+  Args:
+    df: The DataFrame of ratings data.
+    original_users: The original unique user ids in the dataset.
+    original_items: The original unique item ids in the dataset.
+
+  Returns:
+    all_ratings: A set of the [user_id, item_id] with interactions.
+    test_ratings: A list of [user_id, item_id], and each line is the latest
+      user_item interaction for the user.
+    test_negs: A list of item ids with shape [num_users, 100].
+      Each line consists of 100 item ids for the user with no interactions.
+  """
+  # Need to sort before popping to get last item
+  print('Sorting user_item_map by timestamp...')
+  df.sort_values(by='timestamp', inplace=True)
+  all_ratings = set(zip(df[USER_COLUMN], df[ITEM_COLUMN]))
+  user_to_items = collections.defaultdict(list)
+
+  # Generate user_item rating matrix for training
+  t1 = time.time()
+  row_count = 0
+  for row in df.itertuples():
+    user_to_items[getattr(row, USER_COLUMN)].append(getattr(row, ITEM_COLUMN))
+    row_count += 1
+    if row_count % 50000 == 0:
+      print('Processing user_to_items row: {}'.format(row_count))
+  print('Process {} rows in [{:.1f}]s'.format(row_count, time.time() - t1))
+
+  # Generate test ratings and test negatives
+  t2 = time.time()
+  test_ratings = []
+  test_negs = []
+  all_items = set(range(len(original_items)))
+  for user in range(len(original_users)):
+    test_item = user_to_items[user].pop()  # Get the latest one
+
+    all_ratings.remove((user, test_item))  # Remove the test item
+    all_negs = all_items - set(user_to_items[user])
+    all_negs = sorted(list(all_negs))  # determinism
+
+    test_ratings.append((user, test_item))
+    test_negs.append(list(np.random.choice(all_negs, NUMBER_NEGATIVES)))
+
+    if user % 1000 == 0:
+      print('Processing user: {}'.format(user))
+
+  print('Process {} users in [{:.1f}]s'.format(
+      len(original_users), time.time() - t2))
+
+  return all_ratings, test_ratings, test_negs
+
+
+def parse_file_to_csv(data_dir, dataset_name):
+  """Parse the raw data to csv file to be used in model training and evaluation.
+
+  ml-1m dataset is small in size (~25M), while ml-20m is large (~500M). It may
+  take several minutes to process ml-20m dataset.
+
+  Args:
+    data_dir: The directory with the unzipped dataset.
+    dataset_name: The dataset name to be processed.
+  """
+
+  # Use random seed as parameter
```

Comment:
I might be misunderstanding something, but the random seed is being set to 0, not a parameter.